homework-solutions/js/hw-13/task-2.ts

```diff
@@ -0,0 +1,19 @@
+/* Создайте функцию validatePassword, которая принимает строку (пароль) и возвращает true, 
+если пароль соответствует следующим правилам:
+  - Пароль должен содержать хотя бы одну заглавную букву.
+  - Пароль должен содержать хотя бы одну букву в нижнем регистре.
+  - Пароль должен содержать хотя бы одну цифру.
+  - Пароль должен быть не менее 8 символов.
+  - Пароль не должен состоять из одних пробелов
+Функция должна возвращать false, если хотя бы одно из условий не выполнено.
+*/
+
+function validatePassword(password: string): boolean {
+  return !!(
+    password.match(/[A-Z]/) &&
+    password.match(/[a-z]/) &&
+    password.match(/[0-9]/) &&
+    password.length >= 8 &&
+    password.trim().length > 0
+  );
```

Comment:
```suggestion
  return 
    password.match(/[A-Z]/) &&
    password.match(/[a-z]/) &&
    password.match(/[0-9]/) &&
    password.length >= 8 &&
    password.trim().length > 0
```

Оно и так уже вернется булевым, дополнительно приводить не нужно